tests/api_test.py

```diff
@@ -10205,6 +10205,41 @@ def vmap_ref(xs, y):
 
     self.assertEqual(str(jaxpr), str(jaxpr_ref))
 
+  @parameterized.named_parameters(
+    ("1", 1),
+    ("8", 4),
+    ("12", 8),
+    ("16", 16),
+  )
+  def test_batch_map_basic(self, batch_size: int):
+    def f(x):
+      self.assertEqual(x.shape, ())
+      return x**2
+
+    x = np.arange(16)
+    y = jax.lax.batched_vmap(f, batch_size=batch_size)(x)
+
+    np.testing.assert_array_equal(y, x**2)
+
+  @parameterized.named_parameters(
+    ("1", 1),
+    ("8", 4),
+    ("12", 8),
+    ("16", 16),
+  )
+  def test_batch_map_pytrees(self, batch_size: int):
+    f = lambda x: {'b': x['a'] ** 2}
+    inputs = {'a': np.arange(16)}
+    expected = np.arange(16) ** 2
+
+    outputs = jax.lax.batched_vmap(f, batch_size=batch_size)(inputs)
+    np.testing.assert_array_equal(outputs['b'], expected)
+
+    outputs = jax.lax.batched_vmap(
+      f, batch_size=batch_size, in_axes=({"a": 0},), out_axes={"b": -1}
```

Comment:
The in_axes tests here are really limited. We should probably write something combinatorial that checks all cases.

One thing we could do is leave in_axes/out_axes for a follow-up PR, and focus on landing the basic feature first. WDYT?